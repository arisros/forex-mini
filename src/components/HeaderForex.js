import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { baseCurrency } from '../actions/baseCurrency'

export class HeaderForex extends Component {

  constructor() {
    super()
    this.changeBaseCurrencyValue = e => {
      if (parseInt(e.target.value) > -1) {
        this.props.setBaseCurrencyValue(parseInt(e.target.value))
      }
      return false
    };
  }

  componentWillMount() {
    this.props.setBaseCurrency('USD')
    this.props.setBaseCurrencyValue(10)
  }

  render() {
    return (
      <header className="header">
        <p>{this.props.baseCurrency} - United State Dollars</p>
        <div className="sub-header">
          <div>{this.props.baseCurrency}</div>
          <div>
            <input
              className="form-control"
              onChange={this.changeBaseCurrencyValue}
              value={this.props.baseCurrencyValue}
            />
          </div>
        </div>
      </header>
    )
  }
}

const mapStateToProps = (state) => ({
  baseCurrency: state.baseCurrency.baseCurrency,
  baseCurrencyValue: state.baseCurrency.baseCurrencyValue
})

const mapDispatchToProps = (dispatch) => ({
  setBaseCurrency: (currency) => dispatch(baseCurrency.setBaseCurrency(currency)),
  setBaseCurrencyValue: (value) => dispatch(baseCurrency.setBaseCurrencyValue(value))
})

HeaderForex.propTypes = {
  baseCurrency: PropTypes.string,
  baseCurrencyValue: PropTypes.number
};

export default connect(mapStateToProps, mapDispatchToProps)(HeaderForex)
