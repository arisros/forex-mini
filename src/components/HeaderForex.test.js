import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import HeaderForex from './HeaderForex';

configure({ adapter: new Adapter() });

let props, component

beforeAll(() => {

  props = {
    changeBaseCurrencyValue: jest.fn(),
    base: 10,
    baseCurrency: 'USD'
  }

  component = shallow(<HeaderForex {...props} />);
})

describe('HeaderForex Component Snapshot', () => {
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('baseCurrency props  should render correctly', () => {
    expect(component.find('h2').text()).toBe('USD');
  });

  it('base props as a value should render correctly', () => {
    expect(component.find('input').props().value).toBe(10)
  });

  it('changeBaseCurrencyValue props call with', () => {
    component.find('input').simulate('change', 200)
    expect(component
      .find('input')
      .props()
      .onChange
    ).toHaveBeenLastCalledWith(200)
  });

});
