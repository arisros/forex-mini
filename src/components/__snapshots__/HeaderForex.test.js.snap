// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderForex Component Snapshot should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderForex
    base={10}
    baseCurrency="USD"
    changeBaseCurrencyValue={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="sub-header"
      >
        <h2>
          USD
        </h2>
        <div>
          <input
            className="form-control"
            onChange={[MockFunction]}
            value={10}
          />
        </div>
      </div>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            USD
          </h2>,
          <div>
            <input
              className="form-control"
              onChange={[MockFunction]}
              value={10}
            />
          </div>,
        ],
        "className": "sub-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "USD",
          },
          "ref": null,
          "rendered": "USD",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              onChange={[MockFunction]}
              value={10}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "onChange": [MockFunction],
              "value": 10,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="sub-header"
        >
          <h2>
            USD
          </h2>
          <div>
            <input
              className="form-control"
              onChange={[MockFunction]}
              value={10}
            />
          </div>
        </div>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              USD
            </h2>,
            <div>
              <input
                className="form-control"
                onChange={[MockFunction]}
                value={10}
              />
            </div>,
          ],
          "className": "sub-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "USD",
            },
            "ref": null,
            "rendered": "USD",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                onChange={[MockFunction]}
                value={10}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "onChange": [MockFunction],
                "value": 10,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
